src/test/compile-fail/private-in-public-warn.rs

```diff
@@ -246,6 +245,8 @@ mod aliases_priv {
     use self::Priv1 as PrivUseAlias;
     use self::PrivTr1 as PrivUseAliasTr;
     type PrivAlias = Priv2;
+    //~^ WARN private type in public interface
+    //~| WARNING hard error
```

Comment:
`Priv2` is used in a public interface through an alias.
The error location here is suboptimal. On one hand `Priv2` is a true source of the error and pointing to `PrivAlias` is misleading, on the other hand pointing to `Priv2` is confusing too, because the public interface it's used in is not pointed to.
Ideally, there should be two diagnostic messages - one for the public interface location and the other for actual private type location. I'll implement this if positive decision about the whole PR is made.